x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshots.java

```diff
@@ -111,7 +111,8 @@ public void onRepositoriesModule(RepositoriesModule repositoriesModule) {
 
     @Override
     public Optional<EngineFactory> getEngineFactory(IndexSettings indexSettings) {
-        if (SearchableSnapshotRepository.SNAPSHOT_DIRECTORY_FACTORY_KEY.equals(INDEX_STORE_TYPE_SETTING.get(indexSettings.getSettings()))) {
+        if (SearchableSnapshotRepository.SNAPSHOT_DIRECTORY_FACTORY_KEY.equals(INDEX_STORE_TYPE_SETTING.get(indexSettings.getSettings()))
+                && indexSettings.getSettings().getAsBoolean("index.frozen", false) == false) {
```

Comment:
Not sure about introducing this string literal here, vs adding a dependency on the `frozen-indices` module and referring to the setting directly.

Reply:
I would prefer to avoid the dependency and just use the String literal (we have other such cases in our code base).

I also wonder if we need to handle closed replicated snapshot indices (i.e. use `NoOpEngine` for those).

Reply:
We already bypass this mechanism for closed indices:

https://github.com/elastic/elasticsearch/blob/0d1e67dbbbc7de86d1def033786f1cd4283ed4ec/server/src/main/java/org/elasticsearch/indices/IndicesService.java#L622-L627

I added a test case showing that closing and reopening a searchable snapshot index does work.